src/hooks/useDefaultLookup/useDefaultLookup.js

```diff
@@ -0,0 +1,86 @@
+import {
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+
+import {
+  AuthoritiesSearchContext,
+  useFacets,
+} from '@folio/stripes-authority-components';
+
+import {
+  DEFAULT_FILTERS,
+  DEFAULT_LOOKUP_OPTIONS,
+} from '../../constants';
+
+const getNonZeroDefaultFilters = (facets = {}, defaultTagFilters) => {
+  return Object.keys(facets).reduce((acc, facet) => {
+    const nonZeroFilters = defaultTagFilters[facet].filter(defaultFilterId => {
+      return facets[facet].values.some(filter => filter.id === defaultFilterId);
+    });
+
+    if (nonZeroFilters.length) {
+      acc[facet] = nonZeroFilters;
+    }
+
+    return acc;
+  }, { ...DEFAULT_FILTERS });
+};
+
+const useDefaultLookup = (cqlQuery, tag) => {
```

Comment:
I think instead of creating a new hook that uses this context we can add a new prop to the context provider, like `initialValues`
And then we can use data from this object to set initial values inside `AuthoritiesSearchContextProvider`

Reply:
In this case, filters with 0 records will be selected, but we don't need to show them according to the ticket description.

Reply:
We can remove the hook and put all the code into the `SearchModal` component.